plugins/pulp_rpm/plugins/migrations/0031_checksums_and_templates.py

```diff
@@ -0,0 +1,195 @@
+
+    migrate_drpms(drpm_collection)
+
+
+def migrate_drpms(collection):
+    """
+    Add the "checksums" attribute in bulk.
+
+    Since this migration does not need to do any other operations on individual drpms, we'll just
+    add the empty dict. It will be auto-populated as needed during publish operations.
+
+    :param collection:  collection of DRPM units
+    :type  collection:  pymongo.collection.Collection
+    """
+    collection.update_many({}, {'$set': {'checksums': {}}})
+
+
+def migrate_rpm_base(collection, unit):
+    """
+    Templatize the repodata XML and add the "checksums" attribute to the model.
+
+    :param collection:  collection of RPM units
+    :type  collection:  pymongo.collection.Collection
+    :param unit:        the RPM unit being migrated
+    :type  unit:        dict
+    """
+    delta = {}
+    delta['checksums'] = {unit['checksumtype']: unit['checksum']}
+    delta['repodata'] = _modify_xml(unit['repodata'])
+
+    collection.update_one({'_id': unit['_id']}, {'$set': delta})
+
+
+def _modify_xml(repodata):
+    """
+    Given a unit that has repodata XML snippets, modify them in several necessary ways. These
+    include changing the location value and adding template strings to checksum elements.
+
+    This function and everything it calls was copied from pulp_rpm.plugins.db.models, with slight
+    modifications as necessary.
+
+    :param repodata:    the "repodata" attribute from a unit
+    :type  repodata:    dict
+
+    :return:    new repodata dict with templatized XML
+    :rtype:     dict
+    """
+    faked_primary = fake_xml_element(repodata['primary'])
+    primary = faked_primary.find('package')
+
+    faked_other = fake_xml_element(repodata['other'])
+    other = faked_other.find('package')
+
+    faked_filelists = fake_xml_element(repodata['filelists'])
+    filelists = faked_filelists.find('package')
+
+    _templatize_checksum(primary)
+    _templatize_pkgid(other)
+    _templatize_pkgid(filelists)
+
+    return {
+        'primary': remove_fake_element(element_to_text(faked_primary)),
+        'other': remove_fake_element(element_to_text(other)),
+        'filelists': remove_fake_element(element_to_text(filelists)),
+    }
+
+
+def _templatize_pkgid(element):
+    """
+    Modify the passed-in element so that it's "pkgid" element contains a template string
+
+    :param element: XML element that has a "pkgid" attribute
+    :type  element: xml.etree.ElementTree.Element
+    """
+    element.attrib['pkgid'] = '{{ pkgid }}'
+
+
+def _templatize_checksum(package_element):
+    """
+    Modify the checksum element to contain template strings as the text, and as the "type"
+    attribute.
+
+    :param package_element: XML element with name "package" from primary.xml
+    :type  package_element: xml.etree.ElementTree.Element
+    """
+    c_elem = package_element.find('checksum')
+    c_elem.text = '{{ checksum }}'
+    c_elem.attrib['type'] = '{{ checksumtype }}'
```

Comment:
We are specifically following a "RY" pattern for migrations, so I don't think so. They are supposed to be self-contained.